hibernate-reactive-core/src/test/java/org/hibernate/reactive/SQLSelectTest.java

```diff
@@ -0,0 +1,132 @@
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+import static org.hibernate.reactive.testing.DBSelectionExtension.runOnlyFor;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class SQLSelectTest extends BaseReactiveTest {
+	@RegisterExtension // We use native queries, which may be different for other DBs
+	public DBSelectionExtension dbSelection = runOnlyFor( POSTGRESQL );
+
+	private SqlStatementTracker sqlTracker;
+	private Person thePerson;
+
+	// Query containing simple where check to distinguish from a possible generated query
+	private static String selectQuery = "SELECT id, name FROM person WHERE id = $1 and 'hreact' = 'hreact'";
+
+	@Override
+	protected Collection<Class<?>> annotatedEntities() {
+		return List.of( Person.class );
+	}
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		sqlTracker = new SqlStatementTracker( SQLSelectTest::doCheckQuery, configuration.getProperties() );
+		return configuration;
+	}
+
+	private static boolean doCheckQuery(String s) {
+		return s.toLowerCase().startsWith( "select" );
+	}
+
+	@Override
+	protected void addServices(StandardServiceRegistryBuilder builder) {
+		sqlTracker.registerService( builder );
+	}
+
+	@BeforeEach
+	public void populateDb(VertxTestContext context) {
+		List<String> phones = Arrays.asList( "999-999-9999", "111-111-1111", "123-456-7890" );
+		thePerson = new Person();
+		thePerson.id = 724200;
+		thePerson.name = "Claude";
+		thePerson.phones = phones;
+
+		test( context, getMutinySessionFactory().withTransaction( (s, t) -> s.persist( thePerson ) ) );
+	}
+
+	@Test
+	public void findEntity(VertxTestContext context) {
+		test( context, openSession()
+				.thenCompose( session -> session.find( Person.class, thePerson.getId() ) )
+				.thenAccept( found -> {
+					assertPhones( found, "999-999-9999", "111-111-1111", "123-456-7890" );
+					assertThat( sqlTracker.getLoggedQueries() ).contains( selectQuery );
+				} )
+		);
+	}
+
+	private static void assertPhones(Person person, String... expectedPhones) {
+		assertNotNull( person );
+		assertThat( person.getPhones() ).containsExactlyInAnyOrder( expectedPhones );
+	}
+
+	@Entity(name = "Person")
+//	@SQLInsert(sql = "INSERT INTO person (name, id) VALUES ($1, $2) ")
+//	@SQLUpdate(sql = "UPDATE person SET name = $1 where id = $2 ")
+//	@SQLDelete(sql = "UPDATE person SET WHERE id = $1 ")
```

Comment:
Please, remove all the unused commented code

Reply:
removed